spring-integration-core/src/test/java/org/springframework/integration/config/annotation/RouterAnnotationPostProcessorTests.java

```diff
@@ -63,6 +75,24 @@ public void testRouter() {
 		context.stop();
 	}
 
+	@Test
+	public void testRouterWithListParam() {
+		MessagingAnnotationPostProcessor postProcessor = new MessagingAnnotationPostProcessor();
+		postProcessor.setBeanFactory(context.getBeanFactory());
+		postProcessor.afterPropertiesSet();
+		TestRouter testRouter = new TestRouter();
+		postProcessor.postProcessAfterInitialization(testRouter, "test");
+		context.refresh();
+		routingChannel.send(new GenericMessage<String>("foo"));
```

Comment:
Although we are relying on the `ReflectiveMethodExecutor` to convert this to a list of `String`, that would not be obvious to a casual reader who might be trying to understand how things work by reading test cases.

We should either send a `Collections.singletonList("foo")` or make a comment here to help such people.